diff -uBrN qeventdispatcher.orig/qeventdispatcher.pro qeventdispatcher/qeventdispatcher.pro
--- qeventdispatcher.orig/qeventdispatcher.pro	2012-12-19 17:28:49.000000000 +0200
+++ qeventdispatcher/qeventdispatcher.pro	2012-12-25 23:57:54.000000000 +0200
@@ -1,5 +1,8 @@
-CONFIG += testcase
-TARGET = tst_qeventdispatcher
-QT = core testlib
+CONFIG  += testcase
+TARGET   = tst_qeventdispatcher
+QT       = core testlib
 SOURCES += tst_qeventdispatcher.cpp
 DEFINES += QT_DISABLE_DEPRECATED_BEFORE=0
+DESTDIR  = ../
+
+include(../common.pri)
diff -uBrN qeventdispatcher.orig/tst_qeventdispatcher.cpp qeventdispatcher/tst_qeventdispatcher.cpp
--- qeventdispatcher.orig/tst_qeventdispatcher.cpp	2012-12-19 17:28:44.000000000 +0200
+++ qeventdispatcher/tst_qeventdispatcher.cpp	2012-12-26 00:04:20.907752870 +0200
@@ -39,12 +39,9 @@
 **
 ****************************************************************************/
 
-#ifdef QT_GUI_LIB
-#  include <QtGui/QGuiApplication>
-#else
 #  include <QtCore/QCoreApplication>
-#endif
 #include <QtTest/QtTest>
+#include "eventdispatcher_libevent.h"
 
 enum {
     PreciseTimerInterval    =   10,
@@ -209,5 +206,12 @@
     }
 }
 
-QTEST_MAIN(tst_QEventDispatcher)
 #include "tst_qeventdispatcher.moc"
+
+int main(int argc, char** argv)
+{
+    QCoreApplication::setEventDispatcher(new EventDispatcherLibEvent);
+    QCoreApplication app(argc, argv);
+    tst_QEventDispatcher t;
+    return QTest::qExec(&t, argc, argv);
+}
