diff -uBrN qsocketnotifier.orig/qsocketnotifier.pro qsocketnotifier/qsocketnotifier.pro
--- qsocketnotifier.orig/qsocketnotifier.pro	2012-12-19 17:28:49.000000000 +0200
+++ qsocketnotifier/qsocketnotifier.pro	2012-12-25 23:58:27.000000000 +0200
@@ -1,10 +1,12 @@
-CONFIG += testcase
-CONFIG += parallel_test
-TARGET = tst_qsocketnotifier
-QT = core-private network-private testlib
-SOURCES = tst_qsocketnotifier.cpp
+CONFIG  += testcase
+TARGET   = tst_qsocketnotifier
+QT       = core-private network-private testlib network
+SOURCES  = tst_qsocketnotifier.cpp qabstractsocketengine.cpp qnativesocketengine.cpp qnativesocketengine_unix.cpp
+HEADERS  = qabstractsocketengine_p.h qnativesocketengine_p.h
+DEFINES += QT_DISABLE_DEPRECATED_BEFORE=0
+DESTDIR  = ../
 
-requires(contains(QT_CONFIG,private_tests))
+QNETWORK_SRC = $$QT_SOURCE_TREE/src/network
+INCLUDEPATH += $$QNETWORK_SRC
 
-include(../../../network/socket/platformsocketengine/platformsocketengine.pri)
-DEFINES += QT_DISABLE_DEPRECATED_BEFORE=0
+include(../common.pri)
diff -uBrN qsocketnotifier.orig/tst_qsocketnotifier.cpp qsocketnotifier/tst_qsocketnotifier.cpp
--- qsocketnotifier.orig/tst_qsocketnotifier.cpp	2012-12-19 17:28:45.000000000 +0200
+++ qsocketnotifier/tst_qsocketnotifier.cpp	2012-12-26 00:05:05.113845721 +0200
@@ -50,16 +50,11 @@
 #include <QtNetwork/QTcpSocket>
 #include <private/qnativesocketengine_p.h>
 #define NATIVESOCKETENGINE QNativeSocketEngine
-#ifdef Q_OS_UNIX
 #include <private/qnet_unix_p.h>
 #include <sys/select.h>
-#endif
 #include <limits>
 
-#if defined (Q_CC_MSVC) && defined(max)
-#  undef max
-#  undef min
-#endif // Q_CC_MSVC
+#include "eventdispatcher_libevent.h"
 
 class tst_QSocketNotifier : public QObject
 {
@@ -303,5 +298,12 @@
 }
 #endif
 
-QTEST_MAIN(tst_QSocketNotifier)
 #include <tst_qsocketnotifier.moc>
+
+int main(int argc, char** argv)
+{
+    QCoreApplication::setEventDispatcher(new EventDispatcherLibEvent);
+    QCoreApplication app(argc, argv);
+    tst_QSocketNotifier t;
+    return QTest::qExec(&t, argc, argv);
+}
