diff -uwd qtimer/qtimer.pro timer/qtimer.pro
--- qtimer/qtimer.pro	2012-10-22 10:52:27.000000000 +0300
+++ timer/qtimer.pro	2012-11-16 07:19:38.998541453 +0200
@@ -1,5 +1,8 @@
-CONFIG += testcase parallel_test
+CONFIG  += testcase
 TARGET = tst_qtimer
 QT = core testlib
 SOURCES = tst_qtimer.cpp
 DEFINES += QT_DISABLE_DEPRECATED_BEFORE=0
+
+DESTDIR    = ../
+include(../common.pri)
diff -uwd qtimer/tst_qtimer.cpp timer/tst_qtimer.cpp
--- qtimer/tst_qtimer.cpp	2012-10-22 10:48:47.000000000 +0300
+++ timer/tst_qtimer.cpp	2012-11-15 15:47:49.000000000 +0200
@@ -39,13 +39,9 @@
 **
 ****************************************************************************/
 
-#ifdef QT_GUI_LIB
-#  include <QtGui/QGuiApplication>
-#else
 #  include <QtCore/QCoreApplication>
-#endif
-
 #include <QtTest/QtTest>
+#include "eventdispatcher_libevent.h"
 
 #include <qtimer.h>
 #include <qthread.h>
@@ -396,6 +392,7 @@
     ti2.start(MOVETOTHREAD_TIMEOUT);
     QVERIFY((ti1.timerId() & 0xffffff) != (ti2.timerId() & 0xffffff));
     QThread tr;
+    tr.setEventDispatcher(new EventDispatcherLibEvent());
     ti1.moveToThread(&tr);
     connect(&ti1,SIGNAL(timeout()), &tr, SLOT(quit()));
     tr.start();
@@ -717,5 +714,12 @@
     QVERIFY(timerHelper.count > 5);
 }
 
-QTEST_MAIN(tst_QTimer)
 #include "tst_qtimer.moc"
+
+int main(int argc, char** argv)
+{
+	QCoreApplication::setEventDispatcher(new EventDispatcherLibEvent);
+	QCoreApplication app(argc, argv);
+	tst_QTimer t;
+	return QTest::qExec(&t, argc, argv);
+}
